pkg/social/generic_oauth.go

```diff
@@ -264,8 +269,15 @@ func (s *SocialGenericOAuth) extractEmail(data *UserInfoJson) string {
 		return data.Email
 	}
 
-	if data.Attributes["email:primary"] != nil {
-		return data.Attributes["email:primary"][0]
+	var emailAttributeName string
+	if s.emailAttributeName == "" {
+		emailAttributeName = defaultEmailAttributeName
+	} else {
+		emailAttributeName = s.emailAttributeName
+	}
+
+	if data.Attributes[emailAttributeName] != nil {
+		return data.Attributes[emailAttributeName][0]
```

Comment:
Since we have the setting in `defaults.ini` now I think that we can just use `s.emailAttributeName` here as it will always be inherited from there if the user doesn't specify it in `custom.ini`

Reply:
Done